rest/route/build_task.go

```diff
@@ -99,15 +99,12 @@ func (tbh *tasksByBuildHandler) Run(ctx context.Context) gimlet.Responder {
 	for i := range tasks {
 		taskModel := &model.APITask{}
 
-		if err = taskModel.BuildFromService(&tasks[i]); err != nil {
+		if err = taskModel.BuildFromArgs(&tasks[i], &model.APITaskArgs{
+			IncludeAMI: true, IncludeArtifacts: true, IncludeProjectIdentifier: true,
+		}); err != nil {
 			return gimlet.MakeJSONErrorResponder(err)
 		}
-
-		if err = taskModel.GetArtifacts(); err != nil {
-			return gimlet.MakeJSONErrorResponder(err)
-		}
-
-		if err = taskModel.BuildFromService(tbh.url); err != nil {
+		if err = taskModel.BuildFromArgs(tbh.url, nil); err != nil {
```

Comment:
BuildFromService taking in a string at all was kind of hacky. I think now that we have BuildFromArgs it's cleaner to add a parameter for adding logs, and just move 
```
case string:
		ll := LogLinks{
			AllLogLink:    utility.ToStringPtr(fmt.Sprintf(TaskLogLinkFormat, v, utility.FromStringPtr(at.Id), at.Execution, "ALL")),
			TaskLogLink:   utility.ToStringPtr(fmt.Sprintf(TaskLogLinkFormat, v, utility.FromStringPtr(at.Id), at.Execution, "T")),
			AgentLogLink:  utility.ToStringPtr(fmt.Sprintf(TaskLogLinkFormat, v, utility.FromStringPtr(at.Id), at.Execution, "E")),
			SystemLogLink: utility.ToStringPtr(fmt.Sprintf(TaskLogLinkFormat, v, utility.FromStringPtr(at.Id), at.Execution, "S")),
			EventLogLink:  utility.ToStringPtr(fmt.Sprintf(EventLogLinkFormat, v, utility.FromStringPtr(at.Id))),
		}
		at.Logs = ll
```
into that (I think this is especially true if we're going to mark BuildFromService as "deprecated").

Reply:
I moved this over and move all usages to use the buildFromArgs version. There are still some tests that asserted on this value in buildFromService and seemed difficult to move move over. So i left this functionality in buildFromService for those tests.